contracts/DAOTreasury.cdc

```diff
@@ -206,8 +206,12 @@ pub contract DAOTreasuryV2 {
     pub fun depositTokens(identifier: String, vault: @FungibleToken.Vault) {
       emit DepositTokens(identifier: identifier)
 
-      let vaultRef = (&self.vaults[identifier] as &FungibleToken.Vault?)!
-      vaultRef.deposit(from: <- vault)
+      if(self.vaults[identifier] == nil) {
+        self.vaults[identifier] <-! vault
```

Comment:
i think we do not want to let any user deposit a new vault if one doesnt exist for that identifier.  that is an action that only signers should be able to take. i thought we had a `signerDepositVault` function, similar to `signerDepositCollection` but looks like we don't.

i think we should revert this change so that this function will error out if the vault doesnt exist.
created ticket to track [here](https://linear.app/dappercollectives/issue/VSL-139/implement-signerdepositvault-in-daotreasurycdc)

Reply:
So in the end we would need two functions -> one that allows everyone to deposit if the vault exists
And the other one -> where only the signer can add a new vault if it's not existing?

Reply:
If i just revert it it wouldn't be possible to test adding vaults to treasury - either i add signers here or merge an incomplete feature.
I tried adding message signature payload here but it's not feasible because this function is used in `TransferTokenToTreasury -> execute `
and it didn't make sense to add the signatureMessagePayload here because only the signers can execute actions, which would be a duplicate check

Reply:
yes exactly, which i believe is how its working for NFT Collections

Reply:
NFTs: `depositNFT` and `signerDepositCollection`
FTs: `depositTokens` and `signerDepositVault`

Reply:
Updated, can you check again?
In order to add some test coverage for this I'd need to add a new fungible token type to the core contracts and add some tests for it, because all the fungible token vaults are already added through create treasury

Reply:
Added test